Solaris smbios fixes

d101nelson opened the pull request:
### What does this PR do?

Disables 'smbios' execution for SPARC Solaris and attempts to determine smbios related grains:

- serialnumber
- biosversion
- biosversiondate
- uuid
- manufacture
- product

and adds, if available:

- systemfirmware
- systemfirmwaredate

Tested against:

- SPARC Solaris 10 on M5000 hardware
- SPARC Solaris 11 on M3000, M4000, T4-1, T5-4, and T7-2 hardware

### What issues does this PR fix or reference?

Addresses issue  #37613

### Tests written?

No

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html) for best practices.

thusoy committed:
Remove error logging of missing boto libraries

Current behavior always error logs, even if the module is not going to
be used. The loader will raise a proper error if it's attempted
accessed without having loaded properly anyway, and if that happens I'm
assuming you'll turn on debug logging and look for the cause. The
loader also logs which modules are not loaded and their reason, thus
there's no point in repeating the module name in the error message.

m03 committed:
Fix issue 36679 win_servermanager failure

Commit:
allow the file.recurse state to support saltenv (salt://example/dir?saltenv=dev)

Commit:
fixing saltenv if not set in url

Commit:
adding test for saltenv in file.recurse source url

Commit:
making messaging in tests match new return

terminalmage committed:
Strip slashes from gitfs mountpoints

This fixes an issue where mountpoints containing a trailing slash result
in all fileclient requests for files from that repo to fail to match.

terminalmage committed:
Do not use compression in tornado httpclient requests

Tornado releases since 4.0 have a ``decompress_response`` argument
(defaulting to ``True``), which causes any gzip'ed data downloaded in a
tornado HTTPClient request to be decompressed after being downloaded.
Tar archives compressed using gzip therefore end up being decompressed,
which causes errors in ``file.managed`` states as hash verification
fails.

Using ``decompress_response=False`` in the client request resolves this.

This commit also makes the instantiaton of the tornado HTTPClient
instance DRY by not invoking the entire fetch command twice (once if
``max_body_size`` is supported, once if not).

jfindlay committed:
utils.mac_utils: add confirm_updated

jfindlay committed:
modules.mac_system: wait for system to make change

jfindlay committed:
modules.mac_power: wait for system to make change

Commit:
Fix incorrect reference of __utils__ in salt.utils

Closes #37388

Commit:
Auto reconnect `salt` to master if the connection was lost.

1. Re-raise StreamClosedError in the SaltEvent if the event was created
with `raise_errors=True`.
2. Handle StreamClosedError in the client if the client was created with
`auto_reconnect=True`.

twangboy committed:
Create paths.d directory

twangboy committed:
Check for Server os before checking

twangboy committed:
Detect VC for Python correctly on 32bit Windows

Commit:
Cast ip_protocol rule as a str() in boto_secgroup.present

Fixes #36961

Protects against a stacktrace lower in the function where we call
ip_protocol.isdigit(), which expects a string.

Commit:
Write a unit test demonstrating stack trace in #31135

Here is the stack trace that happens when running file.line with
mode=replace on a file that exists, but is empty, as described in
the bug report:

unit.modules.file_test.FileModuleTestCase.test_replace_line_in_empty_file  .................................................
   Traceback (most recent call last):
     File "/root/SaltStack/salt/tests/unit/modules/file_test.py", line 593, in test_replace_line_in_empty_file
       mode='replace'))
     File "/root/SaltStack/salt/salt/modules/file.py", line 1523, in line
       for line in body.split(os.linesep)])
   TypeError: expected a character buffer object

Commit:
file.line with mode=replace on an empty file should return False

Fixes #31135

If a file exists but is empty, we are stack tracing on a call to
the body.split(os.linesep) function. This fix sets the body variable
to an empty string instead of stack tracing when the file is empty.

This allows the function to return ``False``, since no match can be
found in an empty file.

I also adjusted one of the ``line`` variables in the ``replace`` block
because it was shadowing the outer line function scope.

techhat committed:
Make api opts respect correct root_dir

Commit:
Fixup file.line docs to be more clear and consistent

Refs #31081

twangboy committed:
Use cmd.run_all instead of cmd.shell

Commit:
Merge pull request #37459 from twangboy/fix_dsc_json_msg_2016.3

Fix error message when ConvertTo-Json not supported [DO NOT MERGE FORWARD]

Commit:
Merge pull request #37455 from techhat/issue36446

Make api opts respect correct root_dir

Commit:
Merge pull request #37447 from rallytime/fix-36961

Cast ip_protocol rule as a str() in boto_secgroup.present

Commit:
Merge pull request #37446 from twangboy/fix_build_32

Detect VC++ for Python on Win32

Commit:
Merge pull request #37445 from twangboy/fix_import_error_2016.3

Check for Server os before checking [DO NOT MERGE FORWARD]

Commit:
Merge pull request #37442 from twangboy/fix_osx_postinstall

Create paths.d directory

Commit:
Merge pull request #37441 from rallytime/bp-37428

Back-port #37428 to 2015.8

Commit:
Merge pull request #37440 from rallytime/bp-31207

Back-port #31207 to 2016.3

Commit:
Merge pull request #37438 from DSRCorporation/bugs/37238_salt_hang_on_master_restart

Fix for #37238 salt hang on master restart

Commit:
Merge pull request #37426 from jfindlay/mac_sleep

Wait for macOS to change system settings

Commit:
Merge pull request #37418 from terminalmage/issue36849

Do not use compression in tornado httpclient requests

Commit:
Merge pull request #37408 from terminalmage/issue37286

Strip slashes from gitfs mountpoints

Commit:
Merge pull request #37121 from nevins-b/2016.3

allow the file.recurse state to support saltenv

terminalmage committed:
Rewrite file.extract_hash

This rewrite makes it possible for file.extract_hash to match on more
than just the filename being managed, also matching on the source URL as
well as a new user-specified name.

It also makes this function more efficient, as only a single pass is
performed when the hash_type is unspecified. Before, it would read the
entire file for each hash type.

Finally, it adds the ``source_hash_name`` argument to
``file.get_managed``, so that it can be passed to this function by
``file.managed`` states.

terminalmage committed:
Add the source_hash_name param to file.managed states

terminalmage committed:
Update file.extract_hash unit tests

These tests now not only test the new functionality added for matching
on source URI and source_hash_name, but also test non-specific hash_type
lookups, specific hash_type lookups, and failed specific hash_type
lookups (i.e. requesting a hash type not present in the file).

sjorge committed:
improve smartos os version grains

Commit:
Bump log level from debug to warning on empty file

Commit:
Move note about using mode=insert with location options to mode section

sjorge committed:
some more cleanup for smartos

sjorge committed:
improve regex for parsing /etc/release using files from Solaris 8 SPARC and Solaris 10

Commit:
Clarify file.line state docs as well

sjorge committed:
parse minorrelease if it has a / in it

Commit:
Merge branch '2015.8' into '2016.3'

Conflicts:
  - salt/utils/http.py

Commit:
[2015.8] Doc version updated to 2016.3.4 (#37482)

Commit:
[2016.3] Doc version updated to 2016.3.4 (#37483)

Commit:
Get release notes started for 2016.3.5 (#37485)

Commit:
Merge pull request #37478 from rallytime/merge-2016.3

[2016.3] Merge forward from 2015.8 to 2016.3

sjorge committed:
handle Oracle Solaris better

twangboy committed:
Add docs denoting the requirement for at least PowerShell 3

Commit:
Merge branch '2016.3' into 'carbon'

Conflicts:
  - pkg/windows/modules/get-settings.psm1
  - salt/modules/mac_power.py
  - salt/modules/win_dsc.py
  - salt/modules/win_servermanager.py
  - salt/states/file.py
  - salt/utils/event.py
  - salt/utils/gitfs.py

sjorge committed:
detect and properly handle OmniOS

Commit:
Add sdb support to minion and master configs

Commit:
Merge pull request #37472 from sjorge/2016.3-solaris-grains

2016.3 solaris grains improvements

Commit:
Merge pull request #37469 from terminalmage/issue29010

Rewrite file.extract_hash to improve its matching ability

Commit:
Merge pull request #37452 from rallytime/fix-31135

file.line with mode=replace on an empty file should return False, not stacktrace

Commit:
Merge pull request #37493 from cachedout/minion_master_sdb

Add sdb support to minion and master configs

Commit:
Merge pull request #37486 from twangboy/fix_win_docs

Add requirement for PowerShell 3 on Windows

sjorge committed:
Forgot to update os_family map in #37472

mcalmer committed:
fix status handling in sysv init scripts

terminalmage committed:
Update 2016.3.5 release notes with source_hash_name explanation

Commit:
Merge pull request #37497 from terminalmage/release_notes

Update 2016.3.5 release notes with source_hash_name explanation

Commit:
Merge pull request #37496 from mcalmer/fix-status-handling-in-sysv-init-scripts

fix status handling in sysv init scripts

Commit:
Merge pull request #37494 from sjorge/2016.3-osfam_map

Forgot to update os_family map in #37472

Commit:
Merge pull request #37487 from rallytime/merge-carbon

[carbon] Merge forward from 2016.3 to carbon

Commit:
Clarify docs on salt-key events

Closes #37448

Commit:
Remove unused flag

The 'include_errors' option does nothing and has been removed.

Commit:
Log proper message on returners that cannot be loaded

Closes #37444

Commit:
Remove debugging

Commit:
Catch loader error on returners without save_load

Refs #36713

Commit:
Merge pull request #37502 from cachedout/issue_37444

Log proper message on returners that cannot be loaded

Commit:
Merge pull request #37500 from cachedout/remove_include_errors

Remove unused flag

Commit:
Merge pull request #37499 from cachedout/key_docs_clarify

Clarify docs on salt-key events

Commit:
Merge pull request #37503 from cachedout/issue_36713

Catch loader error on returners without save_load

Commit:
Merge branch '2015.8' into '2016.3'

No conflicts.

Commit:
Merge branch '2016.3' into 'carbon'

Conflicts:
  - doc/topics/event/master_events.rst

Commit:
Merge branch 'carbon' into '2016.11'

No conflicts.

Commit:
Update returner __virtual__() return messages for loader

Refs #37444 and #37502

Commit:
Merge pull request #37514 from rallytime/merge-2016.3

[2016.3] Merge forward from 2015.8 to 2016.3

Commit:
Merge pull request #37516 from rallytime/merge-2016.11

[2016.11] Merge forward from carbon to 2016.11

Commit:
Clarify which modes use "before", "after", and "indent" options

This clarification was added both the the file.line module and
file.line state docs.

The-Loeki committed:
http.query(): Properly deprecate 'requests_lib' and honor 'backend'

twangboy committed:
Remove the loop from apply_config

The-Loeki committed:
http.query(): typofix

The-Loeki committed:
modify docs to utilize backend:requests

Commit:
Add syndic_id=None kwarg to save_minions funcs in returners

Fixes #35016

terminalmage committed:
Add missing source_hash_name args to a couple funcs

file.check_managed and file.check_managed_changes both were missing this
new argument. This commit adds it in all places where these functions
are used.

terminalmage committed:
Add missing source_hash_name args to a couple funcs

file.check_managed and file.check_managed_changes both were missing this
new argument. This commit adds it in all places where these functions
are used.

Commit:
Merge pull request #16 from terminalmage/pr-37515

Add missing source_hash_name args to a couple funcs

gtmanfred committed:
fix Lithium to 2015.5.0 (#37530)

twangboy committed:
Remove unnecessary format

The-Loeki committed:
backend:requests - adapt Cloud Azure

Commit:
Merge pull request #37527 from rallytime/fix-35016

Add syndic_id=None kwarg to save_minions funcs in returners

Commit:
Merge pull request #37524 from The-Loeki/opt-backend

http.query(): Properly deprecate 'requests_lib' and honor 'backend'

Commit:
Merge pull request #37519 from rallytime/returner-load-errors

Update returner __virtual__() return messages for loader

The-Loeki committed:
StormPath state: take backend as param for httperrors, deprecate requests_lib:True

whiteinge committed:
Return a 504 response instead of 500 for Salt timeouts

Commit:
Merge pull request #37515 from rallytime/merge-carbon

[carbon] Merge forward from 2016.3 to carbon

Commit:
Merge pull request #37529 from lorengordon/bp-36736

Backport: PR 36736 to 2016.3

The-Loeki committed:
typofix

Commit:
Merge pull request #37533 from whiteinge/salt-api-504-timeouts

Return a 504 response instead of 500 for Salt timeouts

Commit:
Merge pull request #37534 from rallytime/bp-merge-foward-fix

Back-port fix needed from #37515

Commit:
Merge pull request #37526 from twangboy/fix_35799

Remove loop from dsc.apply_config

Commit:
Merge pull request #37457 from rallytime/fix-31081

Fixup file.line docs to be more clear and consistent

Commit:
Merge pull request #37532 from The-Loeki/opt-backend

Backend config: Propagate usage, deprecate requests_lib

Commit:
solaris smbios updates

Commit:
solaris smbios updates

multani committed:
Documentation rendering fixes (#37543)

* doc: fix documentation rendering of salt.runners.survey

* doc: fix Sphinx documentation warnings

Commit:
Don't pass the vpc id to boto.vpc.create_internet_gateway func (#37556)

That function's namespace looks like this:
```
def create_internet_gateway(self, dry_run=False):
```
So when we pass in the vpc_id opbject in the test, the check later
in the function sets `dry_run=True` since the vpc_id opbject exists.

This later throws JSONResponseErrors because the `DryRun` flag is
set. This error raising functionality was added in the most recent
version of moto, which exposed this bug.

This fixes the three boto_vpc_test unit state tests. We'll see if
other tests need to be addressed in other files on a full test run.

Commit:
Merge branch '2016.3' into 'carbon'

Conflicts:
  - doc/topics/releases/releasecandidate.rst
  - salt/modules/win_servermanager.py

Commit:
[PY3] Change log.warn statements to log.warning (#37531)

* [PY3] Change log.warn statements to log.warning

log.warn is deprecated in Python 3. This change removes the
occurance of deprecation warnings for using log.warn. For example:

```
17:24:23 [WARNING ] /testing/salt/utils/verify.py:522: DeprecationWarning: The 'warn' method is deprecated, use 'warning' instead
17:24:23   log.warn('Insecure logging configuration detected! Sensitive data may be logged.')
17:24:23
17:24:23 [WARNING ] Insecure logging configuration detected! Sensitive data may be logged.
```

* Tweak unit.utils.verify_test to using warning instead of warn

Commit:
Merge pull request #37561 from rallytime/merge-carbon

[carbon] Merge forward from 2016.3 to carbon

Commit:
Basic SSL/TLS support in master-minion communication.

Allows to set paths to `ssl_cert` and `ssl_key` in master and minion
opts. Encrypts the TCP channel with the given keys for tcp transport.

Commit:
Update release candidate docs with RC2 info (#37568)

Commit:
Back-port #37569 and #37581 to 2016.11 (#37585)

* Remove known issues section for rc2

All of these have been fixed in the latest rc

* Add smartOS RC build instructions back to RC docs

And update the links to the new packages

Commit:
Merge branch 'carbon' into '2016.11'

Conflicts:
  - salt/modules/file.py

Commit:
Update parameter for building with mock for rpms

Commit:
Merge pull request #37589 from dmurphy18/fix_rpmbuild

Update parameter for building with mock for rpms

Commit:
Merge pull request #37576 from DSRCorporation/features/37449_tls

Basic SSL/TLS support in master-minion communication.

eradman committed:
vmware: optionally allow network interface MAC to be set

Example profile:

    network:
      Network Adapter 1:
        mac: '00:16:3e:e8:19:0f'
        name: vlan25
        switch_type: standard

This takes effect as soon as the VM is initialized, so PXE-booting can
take advantage of this.

Commit:
Remove the `v` from the tag reference (#37595)

When installing with pip, we just need the package version. No `v`.

Commit:
Update the docbuild settings for 2016.11 branch (#37606)

terminalmage committed:
Resolve merge conflict and integrate bugfix with 2016.11 branch

Commit:
Merge pull request #17 from terminalmage/merge-2016.11

Resolve merge conflict and integrate bugfix with 2016.11 branch

Commit:
added manufacture and product; code cleanup

Commit:
added manufacture and product; code cleanup

terminalmage committed:
Raise exception, and also fix error with call to str.format()

Commit:
Merge pull request #18 from terminalmage/merge-2016.11

Raise exception, and also fix error with call to str.format()

Commit:
Merge pull request #37609 from rallytime/merge-2016.11

[2016.11] Merge forward from carbon to 2016.11

Commit:
Merge branch '2016.11' into 'develop'

Conflicts:
  - doc/topics/releases/releasecandidate.rst
  - salt/modules/file.py
  - salt/states/file.py
  - salt/utils/network.py

cachedout commented:
@anlutro Would you like to have a look at this one before we merge this?

@d101nelson It doesn't look like this will merge cleanly. We'll need you to rebase it before we could merge it.

skizunov committed:
Add `info_installed` function to opkg module

This is based on the `info_installed` function in the `zypper` module
in that it accepts an optional extra `attr` kwarg.

One extra feature it has is that it will return information on all
packages if no `names` are specified.

Signed-off-by: Sergey Kizunov <sergey.kizunov@ni.com>

Commit:
Merge pull request #37615 from eradman/vmware

vmware: optionally allow network interface MAC to be set

SolarisYan committed:
if password in profile,not overwrite it

if user provide password in profile,we shouldn't overwrite it

Commit:
Merge pull request #37630 from skizunov/develop2

Add `info_installed` function to opkg module

Commit:
Merge pull request #37631 from SolarisYan/patch-2

if password in profile,not overwrite it

Commit:
cleanup trailing whitespace

Commit:
cleanup trailing whitespace

Commit:
Keep the 'develop' branch setting in doc conf

Commit:
Update "warn" to "warning" in verify_test

Commit:
Merge pull request #37619 from rallytime/merge-develop

[develop] Merge forward from 2016.11 to develop

anlutro commented:
I have no clue about anything this PR mentions so I'm probably not the right person to ask!

Commit:
Merge branch 'solaris_smbios_fixes' of https://github.com/d101nelson/salt into solaris_smbios_fixes

Rebase

cachedout commented:
@anlutro Sorry. My bad!

@d101nelson This rebase didn't go right. Did you rebase against the wrong branch?
